File: model/src/main/java/com/tp/holdem/model/game/Card.java
package com.tp.holdem.model.game;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Card {
	private Suit suit;
	private Honour honour;
	private int value;

	private Card(final Suit suit, final Honour honour) {
		this.suit = suit;
		this.honour = honour;
		this.value = this.honour.value();
	}

	public static Card from(Suit suit, Honour honour) {
		return new Card(suit, honour);
	}

	public static Card coded(String code) {
		return Card.from(Suit.coded(code.charAt(code.length() - 1)), Honour.coded(code.substring(0, code.length() - 1)));
	}

	public String code() {
		return String.format("%s%s", honour.code(), suit.code());
	}
}

File: model/src/main/kotlin/com/tp/holdem/model/Card.kt
package com.tp.holdem.model

import com.tp.holdem.common.model.Honour
import com.tp.holdem.common.model.Suit


data class Card(
        val suit: Suit,
        val honour: Honour,
        val value: Int = honour.value()
) {
    companion object {
        @JvmStatic
        fun from(suit: Suit, honour: Honour): Card {
            return Card(suit, honour)
        }

        @JvmStatic
        fun coded(code: String): Card {
            return from(Suit.coded(code[code.length - 1]), Honour.coded(code.substring(0, code.length - 1)))
        }
    }
}